src/main/java/io/github/mirromutth/r2dbc/mysql/MySqlSslConfiguration.java

```diff
@@ -0,0 +1,241 @@
+        this.keyCertType = requireNonNull(keyCertType, "keyCertType must not be null");
+        this.keyCertUrl = keyCertUrl;
+        this.keyCertPassword = keyCertPassword;
+        this.trustManagerFactory = trustManagerFactory;
+        this.trustCertType = requireNonNull(trustCertType, "trustCertType must not be null");
+        this.trustCertUrl = trustCertUrl;
+        this.trustCertPassword = trustCertPassword;
+    }
+
+    public SslProvider getSslProvider() {
+        return sslProvider;
+    }
+
+    public ClientAuth getClientAuth() {
+        return clientAuth;
+    }
+
+    public String getKeyCertType() {
+        return keyCertType;
+    }
+
+    public String getTrustCertType() {
+        return trustCertType;
+    }
+
+    @Nullable
+    public String[] getTlsProtocols() {
+        return tlsProtocols;
+    }
+
+    @Nullable
+    public KeyManagerFactory getKeyManagerFactory() {
+        return keyManagerFactory;
+    }
+
+    @Nullable
+    public URL getKeyCertUrl() {
+        return keyCertUrl;
+    }
+
+    @Nullable
+    public CharSequence getKeyCertPassword() {
+        return keyCertPassword;
+    }
+
+    @Nullable
+    public TrustManagerFactory getTrustManagerFactory() {
+        return trustManagerFactory;
+    }
+
+    @Nullable
+    public URL getTrustCertUrl() {
+        return trustCertUrl;
+    }
+
+    @Nullable
+    public CharSequence getTrustCertPassword() {
+        return trustCertPassword;
+    }
+
+    static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+
+        private SslProvider sslProvider = SslContext.defaultClientProvider();
+
+        private ClientAuth clientAuth = DEFAULT_CLIENT_AUTH;
+
+        private String keyCertType = DEFAULT_CERT_TYPE;
+
+        private String trustCertType = DEFAULT_CERT_TYPE;
+
+        @Nullable
+        private String[] tlsProtocols;
+
+        @Nullable
+        private KeyManagerFactory keyManagerFactory;
+
+        @Nullable
+        private URL keyCertUrl;
+
+        @Nullable
+        private CharSequence keyCertPassword;
+
+        @Nullable
+        private TrustManagerFactory trustManagerFactory;
+
+        @Nullable
+        private URL trustCertUrl;
+
+        @Nullable
+        private CharSequence trustCertPassword;
+
+        private Builder() {
+        }
+
+        public Builder sslProvider(SslProvider sslProvider) {
+            this.sslProvider = requireNonNull(sslProvider, "sslProvider must not be null");
+            return this;
+        }
+
+        public Builder clientAuth(ClientAuth clientAuth) {
+            this.clientAuth = requireNonNull(clientAuth, "clientAuth must not be null");
+            return this;
+        }
+
+        public Builder disableServerVerify() {
+            return trustManagerFactory(InsecureTrustManagerFactory.INSTANCE);
+        }
+
+        public Builder tlsProtocols(@Nullable String[] tlsProtocols) {
```

Comment:
How about `tlsProtocols(String... protocol)`?

Reply:
Nice, I will treat the empty protocol array same as *default* option (instead of `null` for now), and use `System.arraycopy(…)` to decouple the array.